### DISCUSSION TITLE
How to get it working with copilot.lua (not official copilot.vim)?

### DISCUSSION THREAD

**Discussion by ecosse3** [olimorris/codecompanion.nvim#496] [Q&A] [answered]
Hello,

I would like to use copilot adapter with CodeCompanion but i'm using [zbirenbaum/copilot.lua](https://github.com/zbirenbaum/copilot.lua) library written in Lua for copilot integration instead of official `github/copilot.vim` package. The two conflicts each other with `Copilot` command so I cannot use both.

I get error:
```
Error 401: {"type":"error","error":{"type":"authentication_error","message":"invalid x-api-key"}}
```

Do you know how to handle it and make it work?

On the other hand [CopilotChat.nvim](https://github.com/CopilotC-Nvim/CopilotChat.nvim) works great with lua version of copilot.

**Comment by olimorris** [comment, marked as answer]
It should work straight out of the box. You should just select copilot as the adapter. I presume you've done this?

**Reply by ecosse3** [reply]
I found issue in my config. It's fixed now. I'm posting it below because it may be a bug in the code as well.

```
Error 400: {"type":"error","error":{"type":"invalid_request_error","message":"Your credit balance is too low to access the Anthropic API. Please go to Plans & Billing to upgrade or purchase credits."}}
```

That's my Code Companion config:
```lua
return {
  {
    "olimorris/codecompanion.nvim",
    dependencies = {
      "nvim-lua/plenary.nvim",
      "nvim-treesitter/nvim-treesitter",
      { "MeanderingProgrammer/render-markdown.nvim", ft = { "markdown", "codecompanion" } },
    },
    keys = {
      { "<leader>ccl", "<cmd>CodeCompanion<CR>",     desc = "Inline" },
      { "<leader>ccc", "<cmd>CodeCompanionChat<CR>", desc = "Chat" },
    },
    opts = {
      display = {
        chat = {
          show_settings = true,
        }
      },
      strategies = {
        chat = {
          adapter = "copilot",
        },
        inline = {
          adapter = "copilot",
        },
        agent = {
          adapter = "copilot",
        },
      },
      adapters = {
        copilot = function()
          return require("codecompanion.adapters").extend("anthropic", {
            schema = {
              model = {
                default = "claude-3.5-sonnet",
              },
            },
          })
        end,
      },

    }
  }
}
```

Thie issue was I was extending adapter for "anthropic" instead "copilot" but actually I think it should not show this error because I didn't select "anthropic" as adapter.

Changing that fixed the issue with chat. Anyway the `/buffer` command is not working for me but that's probably another issue and I don't get completions but I guess this plugin is not yet compatible with [blink.cmp](https://github.com/Saghen/blink.cmp).

```diff
- return require("codecompanion.adapters").extend("anthropic", {
+ return require("codecompanion.adapters").extend("copilot", {
```

**Reply by olimorris** [reply]
Great to hear. You can use native completions and use a key map in the chat buffer to trigger it.
